cli/src/main/java/com/devonfw/tools/ide/tool/mvn/Mvn.java

```diff
@@ -248,9 +250,10 @@ public Path getMavenConfFolder(boolean legacy) {
    * @return the maven arguments (MVN_ARGS).
    */
   public String getMavenArgs() {
-    Path mavenConfFolder = getMavenConfFolder(false);
+    Path mavenConfFolder = this.context.getMavenConfigurationFolder();
     Path mvnSettingsFile = mavenConfFolder.resolve(Mvn.SETTINGS_FILE);
-    if (!Files.exists(mvnSettingsFile)) {
+    Path settingsSecurityFile = mavenConfFolder.resolve(SETTINGS_SECURITY_FILE);
+    if (!Files.exists(mvnSettingsFile) && !Files.exists(settingsSecurityFile)) {
```

Comment:
Sorry, if I am picky but now if one of the files does not exist, you are still pointing the a non-existing file and reintroducing the old error we had (do not remember the issue id).
IMHO it is very simple:
```suggestion
    boolean settingsFileExists = Files.exists(mvnSettingsFile);
    boolean securityFileExists = Files.exists(settingsSecurityFile);
    if (!settingsFileExists && !securityFileExists) {
```

Then after the `if` do something like this:
```
  StringBuilder sb = new StringBuilder();
  if (settingsFileExists) {
    sb.append("-s ");
    sb.append(mvnSettingsFile);
  }
  if (securityFileExists) {
    sb.append(" -Dsettings.security=");
    sb.append(settingsSecurityFile.toString().replace("\\", "\\\\"));
  }
  return sb.toString();
```

Entirely delete the method `getSettingsSecurityProperty()` that is anyhow strange since it recomputes what we have already computed in local variables.

Reply:
Adjusted.